python/cuml/cuml/manifold/umap.pyx

```diff
@@ -766,19 +775,33 @@ class UMAP(Base,
         # Handle dense inputs
         else:
             self._sparse_data = False
-            if data_on_host:
+            # Get nnd_n_clusters value for validation
+            build_kwds = self.build_kwds or {}
+            nnd_n_clusters = build_kwds.get("nnd_n_clusters", 1)
+
+            # Validate data_on_host parameter
+            if data_on_host in (True, False):
+                if data_on_host is False and nnd_n_clusters > 1:
+                    raise ValueError(
+                        f"nnd_n_clusters > 1 is not supported when data_on_host is False; "
+                        f"{_DATA_ON_HOST_DEPRECATED_MESSAGE}"
+                    )
+                elif data_on_host is True and nnd_n_clusters == 1:
+                    raise ValueError(
+                        f"nnd_n_clusters == 1 is not supported when data_on_host is True; "
+                        f"{_DATA_ON_HOST_DEPRECATED_MESSAGE}"
+                    )
+                warnings.warn(_DATA_ON_HOST_DEPRECATED_MESSAGE, FutureWarning)
+            elif data_on_host != "auto":
+                raise ValueError(
+                    f"data_on_host must be True, False, or 'auto'; "
+                    f"{_DATA_ON_HOST_DEPRECATED_MESSAGE}"
+                )
+
+            if nnd_n_clusters > 1:
                 convert_to_mem_type = MemoryType.host
```

Comment:
Do we want to warn the user in some way if they pass a device that we have to convert to host? I'm waffling on whether such a warning would be more annoying or more useful to the average user hitting it.